packages/backend/src/telemetry/service.test.ts

```diff
@@ -0,0 +1,47 @@
+import { ConvertError } from '../rates/service'
+import { MockTelemetryService, mockCounter } from '../tests/telemetry'
+
+const telemetryService = new MockTelemetryService()
+describe('TelemetryServiceImpl', () => {
+  it('should create a counter when getOrCreate is called for a new metric', () => {
+    const counter = telemetryService.getOrCreateMetric('testMetric')
+    expect(counter).toBe(mockCounter)
+  })
```

Comment:
I get mocking the telemetry service when testing `collectTelemetryAmount` for example, but doesn't this just test our `MockTelemetryService`, which may or may not be entirely different from our actual `TelemetryServiceImpl`? We could make `TelemetryServiceImpl.getOrCreateMetric` not return a counter and this would still pass. I would expect to test this by creating an actual `TelemetryServiceImpl` and mock the dependencies as needed.

Reply:
valid. Will come back to this ❤️

Reply:
done